### PR TITLE
Add config class to represent the parameters of benchmark script

### PR DESCRIPTION
**Pull request by pzread**
Add config class to load and store all benchmark settings from command arguments. This will help us pass the benchmark settings through functions without having a long list of function parameter.

### CODE REVIEW

File: build_tools/benchmarks/common/benchmark_config.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# All benchmarks' relative path against root build directory.
+
+from argparse import Namespace
+from dataclasses import dataclass
+from typing import Optional
+import os
```
Comment (antiagainst) [inline build_tools/benchmarks/common/benchmark_config.py:11]: Please put `import *` before `from * import *`.

File: build_tools/benchmarks/common/benchmark_config.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+BENCHMARK_SUITE_REL_PATH = "benchmark_suites"
+BENCHMARK_RESULTS_REL_PATH = "benchmark-results"
+CAPTURES_REL_PATH = "captures"
+
+
+@dataclass
+class TraceCaptureConfig:
+  """Represents the settings for capturing traces during the benchamrks.
```
Comment (antiagainst) [inline build_tools/benchmarks/common/benchmark_config.py:20]: ... during benchmarking.

**Review by antiagainst — APPROVED** [review]

File: build_tools/benchmarks/common/benchmark_config.py (hunk 2 above)
Comment (pzread) [reply build_tools/benchmarks/common/benchmark_config.py]: Thanks!

File: build_tools/benchmarks/common/benchmark_config.py (hunk 1 above)
Comment (pzread) [reply build_tools/benchmarks/common/benchmark_config.py]: Done

**Pull request merged**